packages/sdk/src/fee/FeeEstimator.ts

```diff
@@ -0,0 +1,144 @@
+import { Psbt } from "bitcoinjs-lib"
+
+import { AddressFormats, getNetwork, getScriptType } from ".."
+import { Network } from "../config/types"
+import { MAXIMUM_FEE } from "../constants"
+import { FeeEstimatorOptions } from "./types"
+
+export default class FeeEstimator {
+  feeRate: number
+  network: Network
+  psbt!: Psbt
+  witness?: Buffer[] = []
+  fee = 0
+
+  private virtualSize = 0
+  private weight = 0
+
+  constructor({ feeRate, network, psbt, witness }: FeeEstimatorOptions) {
+    if (feeRate < 0 || !Number.isSafeInteger(feeRate)) {
+      throw new Error("Invalid feeRate")
+    }
+
+    this.feeRate = +feeRate // convert decimal to whole number that might have passed Number.isSafeInteger check due to precision loss
```

Comment:
I am not sure how safe is `+`, and how it deals with precision loss, if lets say `feeRate` is 10 and due to precision loss somehow it become 9.9999999999.  I think it might be safer to use `Math.round()`?

Reply:
Case: decimal (in-range / no precision loss / decimal)

number: `1.00000000000001` 

```
Number.isSafeInteger(1.00000000000001) // false
```

The above will not pass the safe integer check and error will be thrown.

----

Case: decimal (not-in-range / precision loss / not-decimal)

number: `1.000000000000000001` 

```
Number.isSafeInteger(1.000000000000000001) // false
```

The above will pass the safe integer check due to precision loss and error will not be thrown. However, since there is loss in precision, JS will consider that number as `1` and ignore the rest of the value in the fraction part. 

----

Numbers is JS are represented in [double-precision floating point](https://en.wikipedia.org/wiki/Double-precision_floating-point_format) format and the ceiled format of the number is strictly equal to the number w/ precision loss. 

```
1.000000000000000001 === 1 // true
```

The being said, the unary operator (+) is redundant because fee rate as a floating number would not pass the check.